x-pack/plugins/spaces/server/lib/request_inteceptors/on_post_auth_interceptor.ts

```diff
@@ -3,40 +3,31 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
+import Boom from 'boom';
+import { PluginProperties, Server } from 'hapi';
+import { Space } from 'x-pack/plugins/spaces/common/model/space';
+import { Feature } from 'x-pack/plugins/xpack_main/types';
+import { wrapError } from '../errors';
+import { getSpaceSelectorUrl } from '../get_space_selector_url';
+import { SpacesClient } from '../spaces_client';
+import { addSpaceIdToPath, getSpaceIdFromPath } from '../spaces_url_parser';
+
+interface XpackMainPlugin {
+  getFeatures: () => Feature[];
+}
 
-import { DEFAULT_SPACE_ID } from '../../common/constants';
-import { wrapError } from './errors';
-import { getSpaceSelectorUrl } from './get_space_selector_url';
-import { addSpaceIdToPath, getSpaceIdFromPath } from './spaces_url_parser';
-
-export function initSpacesRequestInterceptors(server: any) {
-  const serverBasePath = server.config().get('server.basePath');
-
-  server.ext('onRequest', async function spacesOnRequestHandler(request: any, h: any) {
-    const path = request.path;
-
-    // If navigating within the context of a space, then we store the Space's URL Context on the request,
-    // and rewrite the request to not include the space identifier in the URL.
-    const spaceId = getSpaceIdFromPath(path, serverBasePath);
-
-    if (spaceId !== DEFAULT_SPACE_ID) {
-      const reqBasePath = `/s/${spaceId}`;
-      request.setBasePath(reqBasePath);
-
-      const newLocation = path.substr(reqBasePath.length) || '/';
-
-      const newUrl = {
-        ...request.url,
-        path: newLocation,
-        pathname: newLocation,
-        href: newLocation,
-      };
+interface KbnPluginProperties extends PluginProperties {
+  xpack_main: XpackMainPlugin;
+}
 
-      request.setUrl(newUrl);
-    }
+interface KbnServer extends Server {
+  plugins: KbnPluginProperties;
+  getHiddenUiAppById: (appId: string) => any;
+}
 
-    return h.continue;
-  });
+export function initSpacesOnPostAuthRequestInterceptor(server: KbnServer) {
+  const serverBasePath: string = server.config().get('server.basePath');
+  const xpackMainPlugin: XpackMainPlugin = server.plugins.xpack_main;
 
   server.ext('onPostAuth', async function spacesOnRequestHandler(request: any, h: any) {
```

Comment:
nit: this was here before
```suggestion
  server.ext('onPostAuth', async function spacesOnPostAuthHandler(request: any, h: any) {
```